test/models/group_assignment_scopes_test.rb

```diff
@@ -0,0 +1,139 @@
+require 'test_helper'
+
+class GroupAssignmentScopesTest < ActiveSupport::TestCase
+  test 'start_within' do
+    started_before = create_group_assignments 100.days.ago, nil, create(:volunteer)
+    started_before_ended_within = create_group_assignments 100.days.ago, 40.days.ago, create(:volunteer)
+    started_after = create_group_assignments 10.days.ago, nil, create(:volunteer)
+    started_after_ended = create_group_assignments 20.days.ago, 10.days.ago, create(:volunteer)
+    started_within_no_end = create_group_assignments 40.days.ago, nil, create(:volunteer)
+    started_within_ended_within = create_group_assignments 40.days.ago, 35.days.ago, create(:volunteer)
+    started_within_ended_after = create_group_assignments 40.days.ago, 20.days.ago, create(:volunteer)
+    query = GroupAssignment.start_within(45.days.ago..30.days.ago)
+    assert query.include? started_within_no_end.first
```

Comment:
Why `.first`?

Reply:
it returns an array. thats why. needs to be like that